### ISSUE TITLE
[Bug]: java.util.ConcurrentModificationException

### ISSUE THREAD

**Issue by thinhlamzalora** [Appboy/appboy-android-sdk#225]
### Braze Android SDK Version

20.0.0

### Steps To Reproduce

Cannot Reproduce but get a lot of reports from Google Play console

<img width="914" alt="Screen Shot 2022-06-28 at 15 01 19" src="https://user-images.githubusercontent.com/74524101/176126392-a36fcb6f-daac-4929-bfdb-d04017258ab9.png">

### Expected Behavior

not crash anymore

### Actual Incorrect Behavior

this crash is reported by Google console

### Verbose Logs

```shell
java.util.ConcurrentModificationException: 
  at java.util.ArrayList$Itr.next (ArrayList.java:860)
  at com.appboy.Appboy.applyPendingRuntimeConfiguration (SourceFile:3)
  at com.appboy.Appboy.lambda$new$0 (SourceFile:1)
  at com.appboy.Appboy.$r8$lambda$tmujTATmOa-S3_32xNwyFuMla7k (SourceFile)
  at com.appboy.Appboy$$InternalSyntheticLambda$0$934363d0506ef93c8bc054868a7f16e61f0e21230918aa43733dfea8c40e7257$0.run (Appboy.java:4)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1167)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
  at java.lang.Thread.run (Thread.java:929)
```

**Comment by thinhlamzalora** [comment]
hi @radixdev 

Yeah please fix it soon because it affected to a lot of user that using our app

<img width="934" alt="Screen Shot 2022-06-28 at 16 11 34" src="https://user-images.githubusercontent.com/74524101/176330512-b29cb20e-4a71-49a0-900b-e48a52cba922.png">

**Comment by trietbui-zalora** [comment]
I would like to add more info:
- This bug only happens on Huawei/Honor devices: 
- More than 10k crashes.
- 95% on Android 10.
- 100% in background state.

**Comment by trietbui-zalora** [comment]
Hi @radixdev Do you know if the fix will be available soon? 🥲

**Comment by bryanlogan** [comment]
Hi @thinhlamzalora,

We fixed this issue in 22.0.0.  Please let us know if this fixes your issue.

**Comment by thinhlamzalora** [comment]
Hi @bryanlogan 
Thank you very much, we will check and update you later
